### PR TITLE
Add test stubs for Docker feature

### PR DESCRIPTION
**Pull request by elyezer**
Add test stubs for Docker feature on all endpoints. Add stubs for
testing registry for API. For CLI and UI add the same stubs that API
have. Finally for UI add a stub related to the docker tags page.

### CODE REVIEW

File: tests/foreman/cli/test_docker.py [hunk 1]
```diff
@@ -63,3 +75,874 @@ def test_bugzilla_1190122(self):
...
+    @run_only_on('sat')
+    def test_docker_client_upload_image(self):
+        """@Test: A Docker-enabled client can create a new ``Dockerfile``
+        pointing to an existing Docker image from a Satellite 6 and modify it.
+        Then, using ``docker build`` generate a new image which can then be
+        uploaded back onto the Satellite 6 as a new repository.
+
+        @Steps:
+            1.  Publish and promote content view with Docker content
+            2.  Register Docker-enabled client against Satellite 6.
```
Comment (Ichimonji10) [inline tests/foreman/cli/test_docker.py]: @Steps:
    
    1. Publish and promote content view with Docker content.
    2. Register Docker-enabled client against Satellite 6.

File: tests/foreman/ui/test_docker.py [hunk 2]
```diff
@@ -0,0 +1,901 @@
...
+
+    @stubbed()
+    @run_only_on('sat')
+    def test_docker_client_pull_image(self):
+        """@Test: A Docker-enabled client can use ``docker pull`` to pull a
+        Docker image off a Satellite 6 instance.
+
+        @Steps:
+            1.  Publish and promote content view with Docker content
+            2.  Register Docker-enabled client against Satellite 6.
```
Comment (Ichimonji10) [inline tests/foreman/ui/test_docker.py]: List syntax.

File: tests/foreman/ui/test_docker.py [hunk 3]
```diff
@@ -0,0 +1,901 @@
...
+    @run_only_on('sat')
+    def test_docker_client_upload_image(self):
+        """@Test: A Docker-enabled client can create a new ``Dockerfile``
+        pointing to an existing Docker image from a Satellite 6 and modify it.
+        Then, using ``docker build`` generate a new image which can then be
+        uploaded back onto the Satellite 6 as a new repository.
+
+        @Steps:
+            1.  Publish and promote content view with Docker content
+            2.  Register Docker-enabled client against Satellite 6.
```
Comment (Ichimonji10) [inline tests/foreman/ui/test_docker.py]: List syntax.

File: tests/foreman/cli/test_docker.py [hunk 4]
```diff
@@ -63,3 +75,874 @@ def test_bugzilla_1190122(self):
...
+    @run_only_on('sat')
+    @data(
+        gen_string('alpha', 15),
+        gen_string('alphanumeric', 15),
+        gen_string('numeric', 15),
+        gen_string('latin1', 15),
+        gen_string('utf8', 15),
+        gen_string('html', 15),
+    )
+    def test_create_one_docker_repo(self, name):
```
Comment (Ichimonji10) [inline tests/foreman/cli/test_docker.py]: I think many of these methods are missing the `@stubbed` decorator.

**Comment by Ichimonji10** [comment]
ACK pending comments

**Comment by elyezer** [comment]
@Ichimonji10 thanks for the good catches. I have addressed the comments and rebased.

**Comment by omaciel** [comment]
ACK and merge

**Pull request merged**